### PR TITLE
Add support for `javax.cache:cache-api:1.1.1`

### PR DESCRIPTION
**Pull request by linghengqian**
## What does this PR do?

- Fixes https://github.com/oracle/graalvm-reachability-metadata/issues/136 .
- For https://github.com/oracle/graalvm-reachability-metadata/pull/169 and https://github.com/oracle/graalvm-reachability-metadata/pull/124 .

## Checklist before merging
- [x] I have properly formatted metadata files (see [CONTRIBUTING](/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md) document)
- [x] I have added thorough tests. (see [this](/oracle/graalvm-reachability-metadata/blob/master/CONTRIBUTING.md#Tests))

### CODE REVIEW

File: metadata/javax.cache/cache-api/1.1.1/serialization-config.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        "typeReachable": "java.lang.Object"
+      },
+      "name": "java.lang.Number"
+    },
+    {
+      "condition": {
+        "typeReachable": "java.lang.Object"
+      },
+      "name": "java.lang.Integer"
+    }
```
Comment (linghengqian) [inline metadata/javax.cache/cache-api/1.1.1/serialization-config.json:46]: - I was a bit surprised when I manually added these 3 GraalVM reachability metadata entries from nativeTest 's Error Log. This looks like something GraalVM needs to do, but now requires me to add them manually.

**Review by mhalbritter — APPROVED** [review]
Looks good to me.

File: tests/src/javax.cache/cache-api/1.1.1/src/test/java/javax_cache/cache_api/core/stepbystep/SimpleTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            public String process(MutableEntry<String, String> entry,
+                                  Object... arguments) throws EntryProcessorException {
+                return entry.exists() ? entry.getValue().toUpperCase() : null;
+            }
+        });
+        assertThat(result).isEqualTo("BONJOUR");
+        assertThat(cache.get("message")).isEqualTo("bonjour");
+    }
+
+    public static class MyCacheEntryListener implements CacheEntryCreatedListener<String, String>, CacheEntryUpdatedListener<String, String> {
```
Comment (dnestoro) [inline tests/src/javax.cache/cache-api/1.1.1/src/test/java/javax_cache/cache_api/core/stepbystep/SimpleTest.java:55]: Can we use existing MyCacheEntryListener from file declared above?

Comment (linghengqian) [reply tests/src/javax.cache/cache-api/1.1.1/src/test/java/javax_cache/cache_api/core/stepbystep/SimpleTest.java]: - I removed the duplicate in the latest commit. These tests are refactored from the JSR-107 TCK.

**Review by dnestoro — APPROVED** [review]

**Pull request merged**